ast.go

```diff
@@ -2095,14 +2095,19 @@ func resolveReadStatement(node *CustomFuncExpr, resolver ReadStatementResolver)
 		if len(node.Args) != 1 {
 			return "", errors.New("block_num function should have exactly one argument")
```

Comment:
I've added more checks for resolving logic because now we cannot rely on grammar for checking things

Reply:
Cool, now [the comment in the previous PR](https://github.com/tablelandnetwork/go-sqlparser/pull/42#discussion_r1058946781) has the opposite answer. :+1: